src/analysis/normed_space/units.lean

```diff
@@ -0,0 +1,55 @@
+/-
+Copyright (c) 2020 Heather Macbeth. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Heather Macbeth
+-/
+import analysis.normed_space.basic
+import analysis.specific_limits
+import tactic.noncomm_ring
+
+/-!
+# The group of units of a normed ring
+
+In this file we prove that the group of units of a complete normed ring is an open subset of the
+ring.
+-/
+
+noncomputable theory
+
+variables {α : Type*} [normed_ring α] [complete_space α]
+
+/-- In a normed ring, an element `x` of distance less than `1` from `1` is a unit.  Here we
+construct its `units` structure.  -/
+def perturbation_unit (x : α) (h : ∥x∥ < 1) : units α :=
+{ val := 1 - x,
+  inv := tsum (λ (n : ℕ), x ^ n),
+  val_inv := mul_neg_geom_series x h,
+  inv_val := geom_series_mul_neg x h }
+
+lemma unit_of_near_unit [nonzero α] (x : units α) (y : α)
+  (h : ∥y - x∥ < ∥((x⁻¹:units α):α)∥⁻¹) : is_unit y :=
+begin
+  have hpos : 0 < ∥((x⁻¹:units α):α)∥ := norm_pos x⁻¹,
+  have hrad : ∥((x⁻¹:units α):α) * (x - y)∥ < 1,
+  { calc ∥((x⁻¹:units α):α) * ((x:α) - y)∥
+        ≤ ∥((x⁻¹:units α):α)∥ * ∥(x:α) - y∥           : norm_mul_le x.inv _
+    ... = ∥((x⁻¹:units α):α)∥ * ∥y - x∥               : by rw [←neg_sub, norm_neg]
+    ... < ∥((x⁻¹:units α):α)∥ * ∥((x⁻¹:units α):α)∥⁻¹ : by nlinarith [h, hpos]
+    ... = 1                                           : mul_inv_cancel (by linarith [hpos]) },
+  use x * (perturbation_unit _ hrad),
+  unfold perturbation_unit,
+  noncomm_ring, simp,
+end
+
+/-- The group of units of a complete normed ring is an open subset of the ring. -/
+lemma units_open : is_open (is_unit : set α) :=
```

Comment:
This statement works, but I wonder whether it's better to write `{u : α | is_unit u}`. There are too many (sorry!) ways to describe this set, for example also `set.range (coe : units α → α)`.
I think using set notation is the (slightly) canonical choice.

Reply:
Another option is to endow `units α` with a topology, and then show that the `coe` is an open map.
But this requires choosing a topology on `units`. From number theory, I know that the better choice is to view `units` as a subset of `α × α`. But I don't know if this would work in analysis as well...

Reply:
No, we shouldn't impose a topology on units, at least not globally.

Reply:
@PatrickMassot, can you explain?  Putting the restriction of the product topology on the group of units of a topological ring, as @jcommelin suggests, seems reasonable to me; [Wikipedia](https://en.wikipedia.org/wiki/Topological_ring#General_comments) says this will always be a topological group.

Reply:
To be honest I don't remember all details, but we had to deal with that in the perfectoid spaces project. I think at some point we also needed the naive version using the induced topology (which is indeed bad from the point of view of topological groups). However this issue is probably gone in the current version. So let's say I remove my comment. You can add this instance and if it causes problems later then we can downgrade it to a local instance. I guess you'll want to keep your `units_open` statement anyway, and maybe also add version in terms of neighborhoods, but it all depends on what you want to do with this lemma.

Reply:
We can certainly defer a topology on `units` to another PR.
However, I still think that *this* statement should use set notation, instead of coercing a Prop to a set. (Even if it's the same thing.)

Reply:
I made the change to `{u : α | is_unit u}`.  I agree with @jcommelin about what the ideal would be (topological group structure on units by restriction from `α × α`; this theorem becomes that `coe` is an open map), but this can wait to another PR.